ui/components/src/model/units/time.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Duration, milliseconds } from 'date-fns';
+import { UnitGroupConfig, UnitConfig } from './types';
+
+const timeUnitKinds = ['Milliseconds', 'Seconds', 'Minutes', 'Hours', 'Days', 'Weeks', 'Months', 'Years'] as const;
+type TimeUnitKind = typeof timeUnitKinds[number];
+export type TimeUnitOptions = {
+  kind: TimeUnitKind;
+};
+const TIME_GROUP = 'Time';
+export const TIME_GROUP_CONFIG: UnitGroupConfig = {
+  label: 'Time',
+};
+export const TIME_UNIT_CONFIG: Readonly<Record<TimeUnitKind, UnitConfig>> = {
```

Comment:
The `label`s are currently duplicative of the keys in most cases (with a few exceptions in percents). I thought it would be easier to make sense of and maintain the various configs if it was explicit. Let me know if you have a strong preference for only setting the labels when they are different than the keys and falling back to using the key as a name.